python/pyarrow/tests/test_dataset.py

```diff
@@ -2704,6 +2704,24 @@ def test_parquet_dataset_factory(tempdir):
     assert result.num_rows == 40
 
 
+@pytest.mark.parquet
+def test_parquet_dataset_factory_order(tempdir):
+    import pyarrow.parquet as pq
```

Comment:
Add a comment explaining what this tests?

Reply:
I added a comment.